src/Topshelf/TopshelfExitCode.cs

```diff
@@ -12,19 +12,37 @@
 // specific language governing permissions and limitations under the License.
 namespace Topshelf
 {
-    public enum TopshelfExitCode
+    public struct TopshelfExitCode

```

Comment:
Here is the main guy who changed:
1. Enum -> struct
1. Some of the code are combined/replaced
1. Codes are windows error codes compliant